t/151-intercept-log.t

```diff
@@ -0,0 +1,195 @@
+# vim:set ft= ts=4 sw=4 et fdm=marker:
+
+use Test::Nginx::Socket::Lua;
+
+#worker_connections(1014);
+#master_on();
+#workers(2);
+log_level('error');
+
+repeat_each(2);
+
+plan tests => repeat_each() * (blocks() * 3 - 2);
+
+#no_diff();
+#no_long_string();
+run_tests();
+
+__DATA__
+
+=== TEST 1: sanity
+--- http_config
+    lua_intercept_log 4m;
+--- config
+    location /t {
+        access_by_lua_block {
+            ngx.log(ngx.ERR, "enter 1")
+            ngx.log(ngx.ERR, "enter 11")
+
+            local t = ngx.get_log()
```

Comment:
We should only test the FFI-based Lua API provided by `lua-resty-core`. We no longer want to add more CFunction-based Lua API to the global `ngx` table, which can never scale. Everytime we add a new CFunction entry to the `ngx.` global table, the more GC overhead we add at the same time.